bats/tests/containers/allowed-images.bats

```diff
@@ -72,6 +73,7 @@ verify_no_nginx() {
     update_allowed_patterns false '"nginx", "busybox", "ruby"'
     # containerEngine.allowedImages.enabled changed, so wait for a restart
     wait_for_apiserver "$RD_KUBERNETES_PREV_VERSION"
+    wait_for_container_engine
```

Comment:
Hmm, I guess we need to wait for both because waiting for the container engine includes things like buildkit.

It might be clearer if we reverse the order here (wait for the container engine first, _then_ the kube API server).

Reply:
Curiously, it seems like the apiserver is ready before we write the docker context (see #4548).

But I'll change the order, in anticipation of creating the docker context earlier. 😸

Reply:
The API server runs in the VM, so the docker context is irrelevant for its operation.  It can never read the context.

Reply:
I know that the API server doesn't care about the docker context, but the user might. There is no reason for Rancher Desktop to wait for k3s to start successfully before updating the docker context.